src/listener/ArtifactTypeListener.js

```diff
@@ -0,0 +1,57 @@
+import { ToscaArtifactType } from "../model/ToscaArtifactType.js";
+
+export default {
+   exit_artifact_types(parsed_rule) {
+      for (const key in parsed_rule.value) {
+         parsed_rule.value[key].tosca?.setId(
+            key,
+            parsed_rule,
+            "artifact_types"
+         );
+      }
+   },
+
+   // This function is called when the parser encounters a rule
+
+   exit_artifact_type(parsed_rule) {
+      this.checkParsedRule(parsed_rule);
+      const toscaArtifactType = new ToscaArtifactType(
+         this.formatParsedRule(parsed_rule),
+         parsed_rule
+      );
+      parsed_rule.tosca = toscaArtifactType;
+   },
+
+   // This function is called to extract the value of a parsed rule
+   formatParsedRule(parsed_rule) {
+      let properties = new Map();
+      for (const key in parsed_rule.value.properties?.value) {
+         properties[key] = parsed_rule.value.properties.value[key].tosca;
+      }
+      const fileExt = parsed_rule.value.file_ext?.value.map((ele) => ele.tosca);
+      const formattedParsedRule = {
+         derived_from: parsed_rule.value.derived_from?.value,
+         version: parsed_rule.value.version?.tosca,
+         description: parsed_rule.value.description?.value,
+         metadata: parsed_rule.value.metadata?.tosca,
+         properties: properties,
+         mime_type: parsed_rule.value.mime_type?.value,
+         file_ext: fileExt,
+      };
+      return formattedParsedRule;
+   },
+
+   // This function is called to check if the parsed rule is correct
+   checkParsedRule(parsed_rule) {
+      //TODO: Add valid extensions to check file_ext
```

Comment:
should we add an issue on this repo to handle that?

Reply:
I think we will need to yes, the valid extensions are not well defined in the documentation of tosca. I think Xavier has a better idea on the issue.

Reply:
ok thanks